tonic/src/codec/compression.rs

```diff
@@ -157,9 +159,7 @@ impl CompressionEncoding {
         }
     }
 
-    #[allow(missing_docs)]
-    #[cfg(any(feature = "gzip", feature = "zstd"))]
-    pub(crate) fn as_str(&self) -> &'static str {
+    pub(crate) fn as_str(self) -> &'static str {
```

Comment:
I am not sure about the appropriateness of taking ownership where it is not needed.

When taking the ownership, the naming convention suggests to use `into_`.

https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Reply:
Thanks. I have overlooked that it implements `Copy` trait. I am convinced with the performance aspect. Sounds reasonable to use `as_str`.